### PR TITLE
Allow omitting LogRecord in favor of build pattern in Logging SDKs

### PR DESCRIPTION
**Pull request by tigrannajaryan**
The discussion while implementing the Java SDK revealed that the LogRecord
data type may not be necessary. See here https://github.com/open-telemetry/opentelemetry-java/pull/3759#discussion_r738019425

However, in some other languages LogRecord fits nicely (see the comment about
Python implementation in the same comment thread).

This changes allows implementators of the OTEP to choose the most idiomatic
approach for their language.

### CODE REVIEW

**Review by anuraaga — APPROVED** [review]

**Review by lonewolf3739 — APPROVED** [review]

**Review by jkwatson — APPROVED** [review]

**Comment by tigrannajaryan** [comment]
@open-telemetry/specs-approvers I need 2 more approvals on this small amendment please.

**Review by zenmoto — APPROVED** [review]

**Review by jack-berg — APPROVED** [review]

**Comment by tigrannajaryan** [comment]
@open-telemetry/specs-approvers please review.

**Review by jmacd — APPROVED** [review]

**Pull request merged**
